sycl/test/union_param/union_kernel_param.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// TODO: Uncomment once test is fixed on GPU
+// RUNx: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+//==-union_kernel_param.cpp-Checks passing unions as kernel params--------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <stdlib.h>
+
+using namespace cl::sycl;
+
+union TestUnion {
+public:
+  int myint;
+  char mychar;
+  double mydouble;
+
+  TestUnion() { mydouble = 0.0; };
+};
+
+int main(int argc, char **argv) {
+  TestUnion x;
+  x.mydouble = 5.0;
+  bool isError = false;
+
+  cl::sycl::queue queue;
+  queue.submit([&](cl::sycl::handler &cgh) {
+    cgh.single_task<class test>([=]() { x.mydouble; });
```

Comment:
This test is weird. It tests that value that passed by-value, i.e. copied on device wasn't changed :)
Let's change something using this. For example you can create an accessor, write there value of the union member and read data on host.

Reply:
Done